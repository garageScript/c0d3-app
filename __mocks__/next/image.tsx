import React from 'react'
import type { ImageProps } from 'next/image'
import _ from 'lodash'

const Image = (props: ImageProps) => {
  /*not all Image properties are allowed in <img/> tag
  ObjectFit causes warnings, but otherwise lessons/error images won't keep their dimensions 
  see https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit*/
  const copy = _.omit(props, 'objectFit')
  return <img {...copy} />
}

export default Image
