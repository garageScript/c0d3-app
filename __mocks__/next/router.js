const push = jest.fn()
const query = {}
export const useRouter = jest.fn().mockImplementation(() => {
  return {
    route: '/',
    pathname: '/',
    query,
    asPath: '/',
    push: push,
    replace: async () => true,
    reload: () => null,
    back: () => null
  }
})
