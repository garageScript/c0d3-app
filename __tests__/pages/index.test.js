import React from 'react'
import { render, waitFor } from '@testing-library/react'
import { MockedProvider } from '@apollo/client/testing'
import GET_APP from '../../graphql/queries/getApp'
import IndexPage from '../../pages/index'
import dummySessionData from '../../__dummy__/sessionData'
import { useRouter } from 'next/router'

describe('Index Page', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const { push } = useRouter()

  const mocksWithSession = session => [
    {
      request: { query: GET_APP },
      result: {
        data: {
          session,
          lessons: [],
          alerts: []
        }
      }
    }
  ]
  test('Should not redirect on first load', async () => {
    const { push } = useRouter()
    render(
      <MockedProvider mocks={mocksWithSession(null)} addTypename={false}>
        <IndexPage />
      </MockedProvider>
    )
    await waitFor(() => expect(push).not.toHaveBeenCalled())
  })
  test('Should redirect to /curriculum if localstorage loggedIn exists', async () => {
    process['browser'] = true
    window.localStorage.setItem('loggedIn', 'true')
    render(
      <MockedProvider
        mocks={mocksWithSession(dummySessionData)}
        addTypename={false}
      >
        <IndexPage />
      </MockedProvider>
    )
    await waitFor(() => {
      expect(push).toBeCalledWith('/curriculum')
    }) // wait for loading state to pass
  })
})
