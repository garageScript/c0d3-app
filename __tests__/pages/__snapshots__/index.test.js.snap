// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page Should render index page 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand text-primary font-weight-bold"
        href="/"
      >
        C0D3
      </a>
      <div
        id="navbarNav"
      >
        <div
          class="navbar-nav collapse navbar-collapse"
        >
          <div
            class="navbar-nav collapse navbar-collapse"
          >
            <a
              class="nav-item nav-link"
              href="/curriculum"
            >
              Curriculum
            </a>
            <a
              class="nav-item nav-link"
              href="https://github.com/garageScript/c0d3-app"
              rel="noopener noreferrer"
              target="_blank"
            >
              Repo
            </a>
            <a
              class="nav-item nav-link"
              href="/docs/learn"
            >
              Learn
            </a>
            <a
              class="nav-item nav-link"
              href="https://discord.gg/c0d3"
              rel="noopener noreferrer"
              target="_blank"
            >
              Community
            </a>
          </div>
        </div>
      </div>
      <div>
        <a
          class="btn btn-secondary border mr-3"
          href="/login"
        >
          Login
        </a>
        <a
          class="btn btn-secondary border mr-3"
          href="/signup"
        >
          Signup
        </a>
      </div>
    </div>
  </nav>
  <div
    class="bg-white"
  >
    <div
      class="container text-center pb-5"
    >
      <div
        class="row justify-center pt-5"
      >
        <div
          class="col-md-6 offset-md-3"
        >
          <img
            alt="Hero image"
            height="360"
            src="/assets/landing/header-01.svg"
            width="313"
          />
        </div>
      </div>
      <div
        class="row justify-center mt-5"
      >
        <div
          class="col-md-8 offset-md-2"
        >
          <h1
            style="font-weight: 900; font-size: 3.8rem;"
          >
            Become a Software Engineer the Hard Way
          </h1>
        </div>
      </div>
      <div
        class="row justify-center mt-4"
      >
        <div
          class="col-md-10 offset-md-1"
        >
          <p
            class="font-weight-light h4"
            style="line-height: 1.5;"
          >
            Learn the right foundations you need to become a full stack software engineer. Our curriculum takes no shortcuts and effectively trains students with no technical background to become great software engineers.
          </p>
        </div>
      </div>
      <div
        class="row mt-5"
      >
        <div
          class="col-md-6 offset-md-3 mb-5"
        >
          <a
            class="btn btn-primary py-3 px-5"
            href="/curriculum"
          >
            Get Started
          </a>
        </div>
      </div>
      <div
        class="row mt-5"
      >
        <div
          class="col-md-6 offset-md-3"
        >
          <h1
            class="display-4 font-weight-bold"
            id="learning"
          >
            Learning Process
          </h1>
        </div>
      </div>
      <div
        class="row mt-5"
      >
        <div
          class="col-md-10 offset-md-1"
        >
          <p
            class="font-weight-light h4"
            style="line-height: 1.5;"
          >
            Our learning process is interactive and follows the same practice as a well functioning engineering team. From the first line of code you write, you will be code reviewed by engineers to ensure you are always following industry best practices.
          </p>
        </div>
      </div>
      <div
        class="row mt-5 align-items-start mb-5"
      >
        <div
          class="col-md-4"
        >
          <img
            alt="Modern curriculum"
            height="170"
            layout="fixed"
            src="/assets/landing/choose-01.svg"
            width="163"
          />
          <h4
            class="mt-3 font-weight-bold"
          >
            Modern Curriculum
          </h4>
          <p
            class="mt-4 font-weight-light"
          >
            First the curriculum will help you achieve mastery in JavaScript. Using JavaScript, you will learn about different branches of software engineering: websites, mobile applications, servers, and machine learning.
          </p>
        </div>
        <div
          class="col-md-4"
        >
          <img
            alt="Support"
            height="170"
            layout="fixed"
            src="/assets/landing/choose-02.svg"
            width="259"
          />
          <h4
            class="mt-3 font-weight-bold"
          >
            Support
          </h4>
          <p
            class="mt-4 font-weight-light"
          >
            One of the most important skills in a challenging workplace is knowing how to ask for help. Our curriculum is intentionally difficult and we have dedicated mentors to answer your questions and guide you how to communicate effectively.
          </p>
        </div>
        <div
          class="col-md-4"
        >
          <img
            alt="Work Experience"
            height="170"
            layout="fixed"
            src="/assets/landing/choose-03.svg"
            width="93"
          />
          <h4
            class="mt-3 font-weight-bold"
          >
            Work Experience
          </h4>
          <p
            class="mt-4 font-weight-light"
          >
            Once you have completed our curriculum and learned to communicate effectively, you will be invited to contribute to this open-source project to improve the learning experience for future students.
          </p>
        </div>
      </div>
    </div>
    <div
      class="container-fluid py-5"
      style="background: rgb(84, 64, 216); color: white;"
    >
      <div
        class="container mt-5"
      >
        <h1
          class="mt-5 display-4 font-weight-bold text-center"
          id="journey"
        >
          Your Learning Journey
        </h1>
        <p
          class="mt-5 font-weight-light col-md-8 offset-md-2 pb-5 text-center h4"
          style="line-height: 1.5;"
        >
          Our learning process is interactive and follows the same practice as a well functioning engineering team. From the first line of code you write, you will be code reviewed by engineers to ensure you are always following industry best practice
        </p>
        <div
          class="row mt-5 pb-5"
        >
          <div
            class="col-md-5 order-md-2 text-center"
          >
            <img
              alt="Creating an account"
              height="293"
              layout="intrinsic"
              src="/assets/landing/journey-01.svg"
              width="403"
            />
          </div>
          <div
            class="col-md-5 offset-md-1 order-md-1 align-self-center"
          >
            <h2
              class="font-weight-bold"
            >
              Creating an account
            </h2>
            <p
              class="mt-3 font-weight-light"
              style="line-height: 1.5;"
            >
              After creating an account, you will gain access to our code reviewed curriculum and our community chat. You will be able to setup c0d3.com CLI which will allow you to submit code for review. The CLI allows you to code in Javascript from any code editor of your choice.
            </p>
          </div>
        </div>
        <div
          class="row mt-5 pb-5"
        >
          <div
            class="col-md-5 offset-md-1 text-center"
          >
            <img
              alt="Curriculum"
              height="449"
              layout="intrinsic"
              src="/assets/landing/journey-02.svg"
              width="419"
            />
          </div>
          <div
            class="col-md-5 align-self-center"
          >
            <h2
              class="font-weight-bold"
            >
              Curriculum
            </h2>
            <p
              class="mt-3 font-weight-light"
              style="line-height: 1.5;"
            >
              Each lesson in our curriculum is designed to be hands on and you will learn by working through the exercises in the curriculum. All hosting will be provided for you so you can proudly share your accomplishments with the world without spending a penny. The exercises and challenges are designed to be difficult to force you to ask questions. Support is available in our chat channel or in-person at one of our
               
              <a
                class="text-warning"
                href="https://www.meetup.com/San-Jose-C0D3/"
                rel="noopener noreferrer"
                target="_blank"
              >
                meetup groups
              </a>
              . Students who have passed the lessons may be available at the meetup or in the chatroom to help you with your questions.
            </p>
          </div>
        </div>
        <div
          class="row mt-5 pb-5"
        >
          <div
            class="col-md-5 order-md-2 text-center"
          >
            <img
              alt="Pay it forward"
              height="437"
              layout="intrinsic"
              src="/assets/landing/journey-03.svg"
              width="419"
            />
          </div>
          <div
            class="col-md-5 order-md-1 offset-md-1 align-self-center"
          >
            <h2
              class="font-weight-bold"
            >
              Pay It Forward
            </h2>
            <p
              class="mt-3 font-weight-light"
              style="line-height: 1.5;"
            >
              When you finish your lesson, you will have the opportunity to help new students who are starting the lesson. By code reviewing students to the same standard that you were code reviewed, you help ensure that new students are writing good code and becoming better engineers. At the same time, conducting regular code reviews helps you improve your ability to read/debug code as well as understand the lesson concepts at a deeper level.
            </p>
          </div>
        </div>
        <div
          class="row mt-5 pb-5"
        >
          <div
            class="col-md-5 offset-md-1 text-center"
          >
            <img
              alt="Work Experience"
              height="418"
              layout="intrinsic"
              src="/assets/landing/journey-04.svg"
              width="316"
            />
          </div>
          <div
            class="col-md-5 align-self-center"
          >
            <h2
              class="font-weight-bold"
            >
              Work Experience
            </h2>
            <p
              class="mt-3 font-weight-light"
              style="line-height: 1.5;"
            >
              After completing our curriculum, you will gain access to our internal code repository that powers c0d3.com and have the opportunity to work with our engineers and experience the entire workflow of a full stack software engineer. There will be bugs and feature list on our issue board for you to contribute to and you will be held to the same expectations as any respected engineer: high quality code and proper testing. Every feature contribution will be documented and recognized on our contribution list and you would be able to list your contributions as work experience on your resume.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container mt-5 py-5 text-center"
    >
      <h1
        class="display-4 font-weight-bold"
      >
        Start Your Journey
      </h1>
      <p
        class="mt-3 font-weight-light"
        style="line-height: 1.5; font-size: 1.5rem;"
      >
        Become a full stack software engineer - 100% Free!
      </p>
      <a
        class="btn btn-primary py-3 px-5 mt-4"
        href="/curriculum"
      >
        Get Started
      </a>
    </div>
  </div>
  <footer
    class="font-weight-light text-center py-5 bg-white text-muted"
  >
    © Copyright 2021 GarageScript
  </footer>
</div>
`;
